gcc/rust/ast/rust-ast.h

```diff
@@ -1604,11 +1659,15 @@ class ASTFragment
    * ability for a macro to expand to two statements, for instance. */
 
   std::vector<SingleASTNode> nodes;
+  bool fragment_is_error;
 
 public:
-  ASTFragment (std::vector<SingleASTNode> nodes) : nodes (std::move (nodes)) {}
+  ASTFragment (std::vector<SingleASTNode> nodes, bool fragment_is_error = false)
+    : nodes (std::move (nodes)), fragment_is_error (fragment_is_error)
+  {}
```

Comment:
might be worth adding in an assertion hat if the vector of nodes must be empty when the flag fragement_is_error is set